spec/dummy_models_total_count_controller_spec.rb

```diff
@@ -0,0 +1,144 @@
+    let!(:dummy_models) { create_list(:dummy_model, 28) }
+    let(:expected_list) do
+      dummy_models.first(25).map do |dummy|
+        { 'id' => dummy.id, 'name' => dummy.name, 'something' => dummy.something }
+      end
+    end
+
+    context 'when paginating an ActiveRecord with no previous pagination but kaminari installed' do
+      before do
+        get :index
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq expected_list
+      end
+    end
+
+    context 'when paginating with page and limit params' do
+      context 'with a particular limit passed by option' do
+        let(:expected_list) do
+          dummy = dummy_models.third
+          [{ 'id' => dummy.id, 'name' => dummy.name, 'something' => dummy.something }]
+        end
+
+        before do
+          get :index_with_params
+        end
+
+        include_examples 'total count pagination param'
+
+        it 'responds with valid page' do
+          expect(response_body(response)['page']).to eq expected_list
+        end
+      end
+
+      context 'with a really high limit passed by option' do
+        let(:expected_list) do
+          dummy_models.first(50).map do |dummy|
+            { 'id' => dummy.id, 'name' => dummy.name, 'something' => dummy.something }
+          end
+        end
+
+        before do
+          get :index_with_high_limit
+        end
+
+        include_examples 'total count pagination param'
+
+        it 'responds with valid page' do
+          expect(response_body(response)['page']).to eq expected_list
+        end
+      end
+    end
+
+    context 'when paginating an ActiveRecord with a scope' do
+      before do
+        # Requiring both kaminari and will_paginate breaks scope pagination
+        get :index_scoped
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq expected_list
+      end
+    end
+
+    context 'when paginating an ActiveRecord paginated with kaminari' do
+      before do
+        get :index_kaminari
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq expected_list
+      end
+    end
+
+    context 'when paginating an ActiveRecord paginated with will_paginate' do
+      before do
+        get :index_will_paginate
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq expected_list
+      end
+    end
+
+    context 'when paginating an array' do
+      before do
+        get :index_array
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq((1..25).to_a)
```

Comment:
```suggestion
        expect(response_body(response)['page']).to eq([1..25])
```

Reply:
That doesn't work
```
expected: [1..25]
            got: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25]
```
Actually `1..25` is a Range object, so it turns in an array with one Range as the only element